webapp/src/components/Metrics/PerformanceAnalysisTable.tsx

```diff
@@ -0,0 +1,533 @@
+    }
+    const visibility = comparedPipeline !== undefined;
+    const columnVisibilityEntries = [
+      "UtteranceCount",
+      ...ALL_OUTCOMES,
+      ...Object.keys(metricInfo ?? {}),
+      "ece",
+    ].flatMap((field) => [
+      [`comparedPipeline${field}`, visibility],
+      [`delta${field}`, visibility],
+    ]);
+    setColumnVisibilityModel(Object.fromEntries(columnVisibilityEntries));
+  }, [pipeline, comparedPipeline, metricInfo]);
+
+  const getRowSpacing = React.useCallback(({ id }: GridRowSpacingParams) => {
+    return {
+      bottom: id === OVERALL_ROW_ID ? 12 : 0,
+    };
+  }, []);
+
+  const RowLink = (props: RowProps<Row>) => (
+    <Link
+      style={{ color: "unset", textDecoration: "unset" }}
+      to={`/${jobId}/dataset_splits/${selectedDatasetSplit}/performance_overview${constructSearchString(
+        {
+          ...(props.row.id !== OVERALL_ROW_ID && {
+            [selectedMetricPerFilterOption]: [
+              props.row.basePipeline.filterValue,
+            ],
+          }),
+          ...pipeline,
+        }
+      )}`}
+    >
+      <GridRow {...props} />
+    </Link>
+  );
+
+  return errorFetchingBasePipelineData ? (
+    <Typography
+      sx={{ minHeight: 20 }}
+      variant="body2"
+      margin={2}
+      alignSelf="center"
+    >
+      {errorFetchingBasePipelineData.message}
+    </Typography>
+  ) : errorFetchingComparedPipelineData ? (
+    <Typography
+      sx={{ minHeight: 20 }}
+      variant="body2"
+      margin={2}
+      alignSelf="center"
+    >
+      {errorFetchingComparedPipelineData.message}
+    </Typography>
+  ) : (
+    <Paper
+      variant="outlined"
+      sx={{
+        height: "100%",
+        padding: 4,
+        marginTop: 2,
+      }}
+    >
+      <Box marginBottom={1} display="flex" justifyContent="space-between">
+        <Box width={340}>
+          <DatasetSplitToggler
+            availableDatasetSplits={availableDatasetSplits}
+            value={selectedDatasetSplit}
+            onChange={setSelectedDatasetSplit}
+          />
+        </Box>
+        <FormControl size="small">
+          <InputLabel id="compare-pipeline-model">Compared with...</InputLabel>
+          <Select
+            id="compare-pipeline-model-select"
+            value={comparedPipeline ?? "No pipeline"}
+            label="Compared with..."
+            displayEmpty
+            onChange={({ target: { value } }) =>
+              setComparedPipeline(typeof value === "number" ? value : undefined)
+            }
+          >
+            <MenuItem value="No pipeline">
+              <em>No pipeline</em>
+            </MenuItem>
+            {config?.pipelines?.map(
+              (pipelineData, index) =>
+                index !== pipeline.pipelineIndex && (
+                  <MenuItem key={index} value={index}>
+                    {pipelineData.name}
+                  </MenuItem>
+                )
+            )}
+          </Select>
+        </FormControl>
```

Comment:
First, I think we should create a reusable component `PipelineSelect` to make sure both look the same, and avoid duplicating that code about `No pipeline`. Don't worry about the `variant="standard"`, I think it's best to use it, like everywhere else in the app.
Second, I think you are looking for a `FormControlLabel`, not an `InputLabel`.
I think the result should look like
```suggestion
        <FormControlLabel
          control={
            <PipelineSelect
              value={comparedPipeline}
              onChange={setComparedPipeline}
              pipelines={config?.pipelines?.map(
                (_, index) => index !== pipeline.pipelineIndex
              )}
            />
          }
          label="Compared with"
          labelPlacement="start"
          sx={{ gap: 1 }}
        />
```

Reply:
Okay, I don't find "PipelineSelect" in the code. Do you suggest creating one component for both places? I agree its a more efficient way of having it

Reply:
Yes, I said
> we should create a reusable component `PipelineSelect`

Reply:
Yep, I can create one.

Reply:
Any progress on this? Blockers?

Reply:
Yes, I created a separate component for pipeline selection and used it in PageHeader and Performance table.

Reply:
Could you check my suggestion again and try to move the `MenuItem`s inside `PipelineSelect.tsx` as shown? Instead of passing `pipelineMenuItem`, I suggested passing a simple array of pipelines, that I named `pipelines`. In `PageHeader`, that would simply be `pipelines={config.pipelines}`, and in `PerformanceAnalysisTable.tsx`, it would be something like
```ts
              pipelines={config?.pipelines?.filter(
                (_, index) => index !== pipeline.pipelineIndex
              )}
```

Reply:
I tried the solution of putting menuItem inside, but it won't work when we filter the pipeline, and the order of the index gets changed. We need to preserve the right pipeline index to fetch the API result. So, I pass the menuItem dynamically if I need to send filtered pipelines with their associated index.